src/MySqlConnector/MySqlClient/MySqlCommand.cs

```diff
@@ -111,7 +114,70 @@ protected override DbParameter CreateDbParameter()
 
 		internal async Task<int> ExecuteNonQueryAsync(IOBehavior ioBehavior, CancellationToken cancellationToken)
 		{
-			using (var reader = (MySqlDataReader) await ExecuteReaderAsync(CommandBehavior.Default, ioBehavior, cancellationToken).ConfigureAwait(false))
+			if (CommandType == CommandType.StoredProcedure)
+				return await ExecuteStoredProcedureAsync(ioBehavior, cancellationToken);
+
+			return await ExecuteTextCommandNonQueryAsync(CommandText, m_parameterCollection, ioBehavior, cancellationToken);
+		}
+
+		internal async Task<int> ExecuteStoredProcedureAsync(IOBehavior ioBehavior, CancellationToken cancellationToken)
+		{
+			var inParams = new MySqlParameterCollection();
+			var outParams = new MySqlParameterCollection();
+			var argParamNames = new List<string>();
+			var outParamNames = new List<string>();
+			MySqlParameter returnParameter = null;
+			for (var i = 0; i < m_parameterCollection.Count; i++)
+			{
+				var parameter = m_parameterCollection[i];
+				if (parameter.Direction == ParameterDirection.Input || parameter.Direction == ParameterDirection.InputOutput)
+				{
+					inParams.Add(parameter);
+				}
+				if (parameter.Direction == ParameterDirection.Output || parameter.Direction == ParameterDirection.InputOutput)
+				{
+					outParams.Add(parameter);
+					outParamNames.Add(parameter.ParameterName);
+				}
+
+				if (parameter.Direction == ParameterDirection.ReturnValue)
+					returnParameter = parameter;
+				else
+					argParamNames.Add(parameter.ParameterName);
+			}
+
+			int affectedRows;
+			var commandText = "CALL " + CommandText + "(" + string.Join(", ", argParamNames) +")";
```

Comment:
I thought about this too.  I think that all OUT parameters need to be [User Defined Vars](http://dev.mysql.com/doc/refman/5.7/en/user-variables.html), which all begin with `@`.  I think the caller should escape them per documentation if they want them escaped:

> User variables are written as @var_name, where the variable name var_name consists of alphanumeric characters, ., _, and $. A user variable name can contain other characters if you quote it as a string or identifier (for example, @'my-var', @"my-var", or @`my-var`).

Also, sub-queries are technically valid:

`CALL complex((SELECT 1), (SELECT 2), (SELECT "name"), @a, @b, @c, @d, @e)`

This would mean somebody would have to set their IN `MySqlParameter.ParameterName = "(SELECT 1)"`.  They should probably be working with Text Commands at this point anyways though.